gantry/firmware/tim7.c

```diff
@@ -0,0 +1,76 @@
+#include "common/firmware/tim7.h"
+
+#include "common/firmware/errors.h"
+
+TIM_HandleTypeDef htim7;
+
+/**
+ * @brief GPIO Initialization Function
+ * @param None
+ * @retval None
+ */
+void MX_GPIO_Init(void) {
+    /* GPIO Ports Clock Enable */
+    __HAL_RCC_GPIOC_CLK_ENABLE();
+    __HAL_RCC_GPIOF_CLK_ENABLE();
+    __HAL_RCC_GPIOA_CLK_ENABLE();
+    __HAL_RCC_GPIOB_CLK_ENABLE();
+
+    /*Configure GPIO pin : LD2_Pin */
+    GPIO_InitTypeDef GPIO_InitStruct = {0};
+    GPIO_InitStruct.Pin = GPIO_PIN_5;
+    GPIO_InitStruct.Mode = GPIO_MODE_OUTPUT_PP;
+    GPIO_InitStruct.Pull = GPIO_NOPULL;
+    GPIO_InitStruct.Speed = GPIO_SPEED_FREQ_LOW;
+    HAL_GPIO_Init(GPIOA, &GPIO_InitStruct);
+
+    /* EXTI interrupt init*/
+    HAL_NVIC_SetPriority(EXTI15_10_IRQn, 0, 0);
+    HAL_NVIC_EnableIRQ(EXTI15_10_IRQn);
+}
+
+void MX_TIM7_Init(void) {
```

Comment:
Might make sense to have this take the desired interrupt frequency as an argument and do whatever math is required there; failing that, somewhere the frequency should be set in a #define or something with the macro based math from the constants in the system clocking file to derive the prescaler value.

Reply:
I think we'll do this in the next PR. I think we'll be fine-tuning the frequency and setting up the calculations needed to determine whether we should take a step or not.